artifact/metadata.go

```diff
@@ -263,6 +263,31 @@ type ArtifactDepends struct {
 	ArtifactGroup     []string `json:"artifact_group,omitempty"`
 }
 
+type Error struct {
+	Err string
+}
+
+func (e *Error) Error() string {
+	return e.Err
+}
+
+var ErrCompatibleDevices *Error = &Error{"ArtifactDepends: Required field 'CompatibleDevices' not found"}
+
+func (a *ArtifactDepends) UnmarshalJSON(b []byte) error {
+	type Alias ArtifactDepends // Same fields, no inherited UnmarshalJSON method
+	buf := &Alias{}
+	if err := json.Unmarshal(b, buf); err != nil {
+		return err
+	}
+	if len(buf.CompatibleDevices) == 0 {
+		return ErrCompatibleDevices
+	}
```

Comment:
Will this work for v2 artifacts? I'm not sure, but they don't store `CompatibleDevices` in `ArtifactDepends`.

Reply:
Well spotted! Ur right of course. This has to be added.

Reply:
But also, `update-module`s are not supported in `v2`. I will still add the check though.